src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -2838,22 +2853,32 @@ void returnAfter(MethodInformation methodInfo) {
 
         if (methodInfo.returnType == void.class) {
             if (methodInfo.allocator || methodInfo.arrayAllocator) {
-                out.println(indent + "jlong rcapacity = " + (methodInfo.arrayAllocator ? "arg0;" : "1;"));
                 boolean noDeallocator = methodInfo.cls == Pointer.class ||
                         methodInfo.cls.isAnnotationPresent(NoDeallocator.class) ||
                         methodInfo.method.isAnnotationPresent(NoDeallocator.class);
-                out.print(indent + "JavaCPP_initPointer(env, obj, rptr, rcapacity, rptr, ");
-                if (noDeallocator) {
-                    out.println("NULL);");
-                } else if (methodInfo.arrayAllocator) {
-                    out.println("&JavaCPP_" + mangle(methodInfo.cls.getName()) + "_deallocateArray);");
-                    arrayDeallocators.index(methodInfo.cls);
+                if (adapterInfo != null) {
+                    out.println(indent + typeName[0] + " rptr = radapter;");
```

Comment:
If you use typeName[1] as well like `typeName[0] + " rptr" + typeName[1] + " = radapter;"`, does it not work?

Reply:
Yes you must be right, although it seems that `typeName[1]` could differ from `""` only if the object we are allocating is a subclass of `FunctionPointer`